scripts/convert_diffusers_sdxl_lora_to_webui.py

```diff
@@ -0,0 +1,79 @@
+# Script for converting a HF Diffusers trained SDXL LoRAs (be it in the old, new or PEFT format)
+# To the Kohya format used by some WebUIs such as AUTOMATIC1111, ComfyUI, SD.Next and others.
+
+import argparse
+import os
+
+import torch
+from safetensors.torch import load_file, save_file
+
+from diffusers.utils import convert_state_dict_to_peft, convert_unet_state_dict_to_peft
+
+
+def convert_all_diffusers_to_peft(diffusers_dict):
+    try:
+        peft_dict = convert_state_dict_to_peft(diffusers_dict)
+    except Exception as e:
+        if str(e) == "Could not automatically infer state dict type":
+            peft_dict = convert_unet_state_dict_to_peft(diffusers_dict)
+        else:
+            raise
+
+    if not any("lora_A" in key or "lora_B" in key for key in peft_dict.keys()):
+        raise ValueError(
+            "Your LoRA could not be converted to PEFT. Hence, it could not be converted to Kohya/AUTOMATIC1111 format"
+        )
+
+    return peft_dict
+
+
+def convert_peft_to_kohya(state_dict):
+    kohya_ss_state_dict = {}
+    for peft_key, weight in state_dict.items():
+        if "text_encoder_2." in peft_key:
+            kohya_key = peft_key.replace("text_encoder_2.", "lora_te2.")
+        elif "text_encoder." in peft_key:
+            kohya_key = peft_key.replace("text_encoder.", "lora_te1.")
+        elif "unet" in peft_key:
+            kohya_key = peft_key.replace("unet", "lora_unet")
+        kohya_key = kohya_key.replace("lora_A", "lora_down")
+        kohya_key = kohya_key.replace("lora_B", "lora_up")
+        kohya_key = kohya_key.replace(".", "_", kohya_key.count(".") - 2)
+        kohya_ss_state_dict[kohya_key] = weight
+
+        # Set alpha parameter
+        if "lora_down" in kohya_key:
+            alpha_key = f'{kohya_key.split(".")[0]}.alpha'
+            kohya_ss_state_dict[alpha_key] = torch.tensor(len(weight))
+
+    return kohya_ss_state_dict
+
+
+def convert_and_save(input_lora, output_lora=None):
+    if output_lora is None:
+        base_name = os.path.splitext(input_lora)[0]
+        output_lora = f"{base_name}_webui.safetensors"
+
+    diffusers_state_dict = load_file(input_lora)
+    peft_state_dict = convert_all_diffusers_to_peft(diffusers_state_dict)
+    kohya_state_dict = convert_peft_to_kohya(peft_state_dict)
+    save_file(kohya_state_dict, output_lora)
```

Comment:
Do we wanna have an option for `push_to_hub`? That'd be very cool IMO.

Reply:
Love it! I'd merge it as is for now and add it in a next PR , we can also have other utils to PR a Kohya weight to existing diffusers trained repos and other cool utils like that